### PR TITLE
revisions from the first referee report (related to "space-filling")

### PR DESCRIPTION
**Pull request by hegdevinayi**
Addresses one of the referee comments about comparing model accuracy with MU vs a `space-filling` approach.

- The sequential learning workflow now implements a Euclidean distance-based aggregation of the most-distant candidate (to the current training set, in the feature vector space) in each SL iteration.
- Adds a separate plotting script that prepares a figure comparing all four acquisition functions (MLI, MU, random, and space-filling), to be added to the SI.
- The previous plotting script now adds random to the model accuracy plot (panel 3 in the figure).

### CODE REVIEW

File: scripts/sequential_learning.py [hunk 1]
```diff
@@ -120,6 +140,20 @@ def choose_next_candidate(
...
+        feature_columns = [c for c in df.columns if c not in non_feature_columns]
+        X = np.array(df[feature_columns].values)
+        # scale the data (for using distance-based metrics)
+        scaler = StandardScaler()
+        X = scaler.fit_transform(X)
+        # calculate all pairwise distances between examples (feature vectors)
+        D = euclidean_distances(X, X)
+        # print(f"D.shape: {D.shape}")
+        # choose the candidate in the design space that is farthest away from the
+        # training set
```
Comment (lancekavalsky) [inline scripts/sequential_learning.py]: can we rephrase slightly to try and make the logic of the below
lines a little more obvious? maybe something like below?
```suggestion
        # choose the unlabelled candidate in the design space whose 
        # closest neighbor in the training set is the farthest away
```

File: scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+    target_window: List[float] = None,
+    *,
+    n_iterations: int = None,
+    ax: Axes = None,
+):
+    """
+    Plot the average candidate distance to the target window as a function of SL
+    iteration, and color an area around the average to indicate uncertainty (calculated
+    as +/- std over independent SL trials). Compare the performance of maximum
+    likelihood of improvement (MLI) and random acquisition functions.
```
Comment (lancekavalsky) [inline scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py]:
```suggestion
    as +/- std over independent SL trials). Compare the performance of maximum
    likelihood of improvement (MLI), maximum uncertainty (MU), space-filling, 
    and random acquisition functions.
```

File: scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+    target_window: List[float] = None,
+    *,
+    n_iterations: int = None,
+    ax: Axes = None,
+):
+    """
+    Plot the average number of candidates found in the target window as a function of SL
+    iteration, and color an area around the average to indicate uncertainty (calculated
+    as +/- std over independent SL trials). Compare the performance of maximum
+    likelihood of improvement (MLI) and random acquisition functions.
```
Comment (lancekavalsky) [inline scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py]:
```suggestion
    as +/- std over independent SL trials). Compare the performance of maximum
    likelihood of improvement (MLI), maximum uncertainty (MU), space-filling, 
    and random acquisition functions.
```

File: scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+    /,
+    target_column: str = "binding_energy_of_adsorbed",
+    *,
+    init_train_size: int = 10,
+    n_iterations: int = None,
+    ax: Axes = None,
+):
+    """
+    Plot model accuracy as a function of SL iteration for the maximum uncertainty (MU)
+    acquisition function, and color an area around the average to indicate uncertainty
```
Comment (lancekavalsky) [inline scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py]:
```suggestion
    Plot model accuracy as a function of SL iteration for, maximum uncertainty (MU),
    maximum likelihood of improvement (MLI), space-filling, and random
    acquisition functions, and color an area around the average to indicate uncertainty
```

**Review by lancekavalsky — COMMENTED** [review]
some minor suggestions mainly related to documentation and code comments

File: scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py [hunk 5]
```diff
@@ -0,0 +1,639 @@
+"""
+Reproduce "acceleration from sequential learning"-related figures from the associated
+publication (corresponding to Figure 5).
+
+Note that this script is provided for reproduction purposes only: the data aggregation,
+averaging, and plotting functionalities implemented here contain several hard-coded,
+problem-specific parameters. Please do not expect these functions to work as-is without
+modifications for generating similar figures for a different problem.
+
+The `make_figure` function encapsulates the overall task of generating Figure 5, with
```
Comment (lancekavalsky) [inline scripts/plot_acceleration_from_sequential_learning__ALL_ACQ.py]:
```suggestion
The `make_figure` function encapsulates the overall task of generating Figure S1, with
```

**Review by lancekavalsky — APPROVED** [review]
